cmd/geth/snapshot.go

```diff
@@ -1020,3 +1036,213 @@ func checkAccount(ctx *cli.Context) error {
+		CacheSize:  256,
+		Recovery:   false,
+		NoBuild:    true,
+		AsyncBuild: false,
+	}
+	triesInMemory := ctx.Uint64(utils.TriesInMemoryFlag.Name)
+	snaptree, err := snapshot.New(snapConfig, db, tdb, root, int(triesInMemory), false)
+	snap := snaptree.Snapshot(root)
+	if err != nil {
+		return err
+	}
+
+	// init key
+	keySet := make([]common.Hash, 0, 1000)
+	for i := 0; i < 1000; i++ {
+		key := common.BigToHash(big.NewInt(int64(i)))
+		keySet = append(keySet, crypto.Keccak256Hash(key.Bytes()))
+	}
+
+	cpuCount := ctx.Int64(utils.CPUCountFlag.Name)
+	p := NewRoutinePool(cpuCount*100, snap, keySet)
+
+	it := db.NewIterator(rawdb.SnapshotAccountPrefix, nil)
+	defer it.Release()
+
+	// Inspect key-value database first.
+	idx := uint64(0)
+	for it.Next() {
+		key := it.Key()
+		switch {
+		case bytes.HasPrefix(key, rawdb.SnapshotAccountPrefix) && len(key) == (len(rawdb.SnapshotAccountPrefix)+common.HashLength):
+			var account types.SlimAccount
+			err := rlp.DecodeBytes(it.Value(), &account)
+			if err != nil {
+				log.Error("failed to DecodeBytes", "addrHash", key[1:])
+				continue
+			}
+			if bytes.Equal(account.CodeHash, types.EmptyCodeHash.Bytes()) {
+				continue
+			}
+			if idx%10000 == 0 {
+				fmt.Printf("add task: %d\n", idx)
+			}
+			idx++
+			p.AddTask(common.BytesToHash(key[1:]))
+		}
+	}
+
+	p.Wait()
+	fmt.Println("all done")
+	fmt.Println("var count", p.varCount.Load())
+	fmt.Println("arr count", p.arrCount.Load())
+	fmt.Println("arr item count", p.arrItemCount.Load())
+
+	f, err := os.Create("result.txt")
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	_, err = f.WriteString(fmt.Sprintf("var count: %d\n", p.varCount.Load()))
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	_, err = f.WriteString(fmt.Sprintf("arr count: %d\n", p.arrCount.Load()))
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	_, err = f.WriteString(fmt.Sprintf("arr item count: %d\n", p.arrItemCount.Load()))
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	_, err = f.WriteString(fmt.Sprintf("empty contract count: %d\n", p.emptyContract.Load()))
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	arrCount := make([]int, 0, len(p.arrItemCountMap))
+	for k := range p.arrItemCountMap {
+		arrCount = append(arrCount, k)
+	}
+	sort.Ints(arrCount)
+	sum := uint64(0)
+	maxCount := uint64(0)
+	mode := 0
+	for _, k := range arrCount {
+		count := p.arrItemCountMap[k]
+		sum += count
+		if sum >= p.arrCount.Load()/2 {
+			fmt.Printf("median: %d\n", k)
+			_, err = f.WriteString(fmt.Sprintf("median: %d\n", k))
+			if err != nil {
+				fmt.Println(err)
+			}
+			break
+		}
+	}
+
+	for _, k := range arrCount {
+		count := p.arrItemCountMap[k]
+		if maxCount < count {
+			maxCount = count
+			mode = k
+		}
+	}
+
+	fmt.Printf("mode: %d\n", mode)
+	_, err = f.WriteString(fmt.Sprintf("mode: %d\n", mode))
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	storageIt := db.NewIterator(rawdb.SnapshotStoragePrefix, nil)
+	defer storageIt.Release()
+
+	storageKeyCount := uint64(0)
+	for storageIt.Next() {
```

Comment:
please add this condition `len(key) == (len(SnapshotStoragePrefix)+2*common.HashLength)`

Reply:
fixed